src/clusterfuzz/_internal/system/archive.py

```diff
@@ -197,137 +339,28 @@ def unpack(archive_path,
-        # to chmod it explicitly. Also, get rid of suid bit for security
-        # reasons.
-        external_attr = zip_archive.getinfo(filename).external_attr >> 16
-        if oct(external_attr).startswith(FILE_ATTRIBUTE):
-          old_mode = external_attr & 0o7777
-          new_mode = external_attr & 0o777
-          new_mode |= 0o440
-          needs_execute_permission = external_attr & 100
-
-          if new_mode != old_mode or needs_execute_permission:
-            # Default extract condition is 640 which is safe.
-            # |new_mode| might have read+write+execute bit for
-            # others, so remove those.
-            new_mode &= 0o770
-
-            os.chmod(extracted_path, new_mode)
-
-        # Create symlink if needed (only on unix platforms).
-        if (trusted and hasattr(os, 'symlink') and
-            oct(external_attr).startswith(SYMLINK_ATTRIBUTE)):
-          symlink_source = zip_archive.read(filename)
-          if os.path.exists(extracted_path):
-            os.remove(extracted_path)
-          os.symlink(symlink_source, extracted_path)
-
-        # Keep heartbeat happy by updating with our progress.
-        archive_file_unpack_count += 1
-        if archive_file_unpack_count % 1000 == 0:
-          logs.log('Unpacked %d/%d.' % (archive_file_unpack_count,
-                                        archive_file_total_count))
-
-      except:
-        # In case of errors, we try to extract whatever we can without errors.
-        error_occurred = True
-        continue
-
-    logs.log('Unpacked %d/%d.' % (archive_file_unpack_count,
-                                  archive_file_total_count))
-    zip_archive.close()
-    zip_file_handle.close()
-
-    if error_occurred:
-      logs.log_error(
-          'Failed to extract everything from archive %s.' % archive_filename)
-
-  elif archive_type in (ArchiveType.TAR, ArchiveType.TAR_LZMA):
-    if archive_type == ArchiveType.TAR_LZMA:
-      lzma_file = lzma.LZMAFile(archive_path)
-      tar_archive = tarfile.open(fileobj=lzma_file)
-    else:
-      tar_archive = tarfile.open(archive_path)
-
-    try:
-      tar_archive.extractall(path=output_directory)
-    except:
-      # In case of errors, we try to extract whatever we can without errors.
-      error_occurred = True
-      logs.log_error(
-          'Failed to extract everything from archive %s, trying one at a time.'
-          % archive_filename)
-      for filename in file_list:
-        try:
-          tar_archive.extract(filename, output_directory)
-        except:
-          continue
-
-        # Keep heartbeat happy by updating with our progress.
-        archive_file_unpack_count += 1
-        if archive_file_unpack_count % 1000 == 0:
-          logs.log('Unpacked %d/%d.' % (archive_file_unpack_count,
-                                        archive_file_total_count))
-
+  error_occurred = False
+  for file in file_list:
+    error_occurred = reader.extract(
```

Comment:
This should likely be:

```suggestion
    error_occurred |= reader.extract(
```